tests/src/test/scala/com/nvidia/spark/rapids/RapidsBufferCatalogSuite.scala

```diff
@@ -320,6 +324,125 @@ class RapidsBufferCatalogSuite extends AnyFunSuite with MockitoSugar {
     verify(buffer3).free()
   }
 
+  test("concurrently acquiring buffer and registering new buffer with the same ID") {
```

Comment:
My understanding is that the new buffer is registered when `updateTiers` is called which in turn can be called for the spilling and the unspilling. Am I missing something?

Reply:
> My understanding is that the new buffer is registered when updateTiers is called which in turn can be called for the spilling and the unspilling.

This is correct, but what is the problem with that exactly? That you may add a reference to a buffer that is about to move from device to host at the same time as it is spilling. The spill framework was first created without many locks, the idea being that these kinds of race conditions was OK. So, we will be eventually consistent. Now if we find that there is an issue where we are OOMing because we are not locking things enough, I see us revisiting that. Does this make sense?